core/txpool/legacypool/legacypool.go

```diff
@@ -1520,12 +1556,30 @@ func (pool *LegacyPool) reset(oldHead, newHead *types.Header) {
 	pool.currentState = statedb
 	pool.pendingNonces = newNoncer(statedb)
 
+	// we don't care about these
 	// Inject any transactions discarded due to reorgs
 	log.Debug("Reinjecting stale transactions", "count", len(reinject))
 	core.SenderCacher.Recover(pool.signer, reinject)
 	pool.addTxsLocked(reinject, false)
 }
 
+// reset retrieves the current state of the blockchain and ensures the content
+// of the transaction pool is valid with regard to the chain state.
```

Comment:
>ensures the content of the transaction pool is valid with regard to the chain state.

this looks like it just sets the pool's local state, but doesn't actually change anything regarding txs the way `reset` does. it that intended? are the txs validated and updated somewhere else?